### ISSUE TITLE
Repeater fields within repeater fields not deleted

### ISSUE THREAD

**Issue by germanny** [elliotcondon/acf#411]
We've discovered a problem with nested repeater field data not being deleted when a nested repeater row is deleted.

To replicate the issue, create a repeater field that has a maybe couple text fields and a repeater field. Add a row or two, and then delete a row. Check the database and see all the fields except the repeater fields that had data in them are deleted. However, the nested repeater field's data remains, and if you create a new row, that data creeps into that new row (because they have the same identify number attached) unless you populate that field with new data.

So in other words, that new row's repeater field cannot be blank, even though it doesn't actually show any data in the interface.

We noticed the issue in 5.2.6 and 5.2.7.

Thanks!
Jen G

**Comment by elliotcondon** [comment]
Hi @germanny 

Thanks for the bug report.

Currently, the repeater field does not listen for the `acf/delete_value` filter which explains why the nested repeater field values are not cleared.

I can add this in, however, I'm not sure I understand how the values creep back in. I'll do some testing, but if you can post some screenshots of the issue, that would be great!

**Comment by germanny** [comment]
Hey @elliotcondon 

Thanks for the fast response. I'll do my best to give you screenshots that make sense.

* Here's a shot of the field as I see it in `/wp-admin`: http://j.mp/1fKO8OK.
* Here's a shot of the data populating this field in the database: http://j.mp/1fKOftr. Notice the huge discrepancy in the meta_id values; this gave us an idea that some of the rows had been created while ago.
* Here's a shot of the database again, this time after I delete that row from `/wp-admin`: http://j.mp/1fKOtAY. All those old ids still exist.
* When I create a new row, the new data gets the `_1_` id in the `meta_key` field, so as I pull the data from the database, it gets pulled in even though we can't see it in `/wp-admin`. http://j.mp/1fKQpJD

Here is the export file of the fields for this: http://j.mp/1fKOUuR.

Hope that helps. Please let me know what questions I can help answer.

Thanks!
Jen

**Comment by elliotcondon** [comment]
Hi @germanny 

When you create a new row via the admin interface, you will $_POST all the sub field values for that row.
This will override all the sub field values - which contradicts what your screenshots show.

Perhaps you are saving data to a different post_id than what you are viewing in the DB?

**Comment by germanny** [comment]
Well, no, I'm sorry but that's not happening. :( The 3 screenshots of the database, taken in order from original data, then deleted but showing the nested repeater "tuition_#_label" and "tuition_#_amount" still exists, then added again to the same post, all show the same post_ID. If there had been a subfield with ID of 1 (in the example shown in the screenshots) and that is deleted and later another row added, it assumes the data that wasn't deleted from the original removal. Nested repeater field data is *not* deleted. We are overcoming this (hopefully temporarily) by requiring a minimum number of rows and setting the default value to 0. This helps if there was only one row in the nested repeater, but additional rows are not overwritten so the data still shows.

This is a relatively new issue we started noticing within the past couple of releases.

Please let me know if this clears things up a bit.

Thanks!
Jen

**Comment by elliotcondon** [comment]
Hi @germanny 

Thanks for the info.

I think I see the issue now.
So when you add a new row, you leave the sub field (repeater) blank. 
When you save the post and re-load the page, the previous value is loaded back in?

I'll do some testing with this and get back to you

Thanks
E

**Comment by germanny** [comment]
Hi @elliotcondon:

Thanks for all your help with this.

I think you have it now. :) Subfields are not deleted when they are nested in another subfield. So if I have:

* Subfield (repeater) with id 1
  * Child field (not repeater) with id 1
  * Child subfield (repeater) with id 1

If I delete Subfield 1, Child field 1 is deleted but Child subfield 1 is not deleted (as it should be). So it remains with id 1.

If I add another Subfield which gets an id of 1 because the previous Subfield with id 1 was deleted, and if I do not populate the Child subfield with the same amount of data, then the Subfield 1 assumes the Child subfield 1 data that still exists in the database.

It doesn't show them in the UI in `wp-admin` but programmatically you can pull them via an SQL query into your dataset.

Thanks!
Jen

**Comment by elliotcondon** [comment]
Hi @germanny 

Thanks for the info. 

Yep, I'll do some testing soon and add in a fix for this

Thanks
E

**Comment by elliotcondon** [comment]
Hi @germanny 

Thanks again for your help with this issue.
I've successfully added in some new logic to clean up old sub field values and will include it into the next version

Thanks
E

**Comment by ukeboyska** [comment]
@elliotcondon i made sure i'm using the latest version of ACF but this seems to be happening again... i have several flexible content sub fields within a repeater field (a user picks the type of flexible content type the user wants to add with a dropdown select), but when i remove any of the flexible content rows using the minus icon they still display on the front end as if they were never deleted. any ideas?

**Comment by elliotcondon** [comment]
Hi @ukeboyska 

Thanks for the reply. Would you kindly export your field group to .json and attach it to a reply so I can test it locally?

**Comment by ukeboyska** [comment]
thanks for the quick reply! couldn't attach the file so here's the code:

`[
    {
        "key": "group_57e97ce1d6a3b",
        "title": "Customizable Content",
        "fields": [
            {
                "key": "field_57e97d06b7854",
                "label": "Custom Content Repeater",
                "name": "custom_content_repeater",
                "type": "repeater",
                "instructions": "",
                "required": 0,
                "conditional_logic": 0,
                "wrapper": {
                    "width": "",
                    "class": "",
                    "id": ""
                },
                "collapsed": "",
                "min": "",
                "max": "",
                "layout": "block",
                "button_label": "Add Content Section",
                "sub_fields": [
                    {
                        "key": "field_57e97ff963e85",
                        "label": "Custom Content Picker",
                        "name": "custom_content_picker",
                        "type": "select",
                        "instructions": "Choose a type of content to add next",
                        "required": 1,
                        "conditional_logic": 0,
                        "wrapper": {
                            "width": 75,
                            "class": "",
                            "id": ""
                        },
                        "choices": {
                            "---": "---",
                            "Icon & Content": "Icon & Content",
                            "Image w\/ Caption": "Image w\/ Caption",
                            "WYSIWYG": "WYSIWYG",
                            "50\/50 List": "50\/50 List",
                            "Button": "Button"
                        },
                        "default_value": [],
                        "allow_null": 0,
                        "multiple": 0,
                        "ui": 0,
                        "ajax": 0,
                        "placeholder": "",
                        "disabled": 0,
                        "readonly": 0,
                        "return_format": "value"
                    },
                    {
                        "key": "field_57e9808a40d79",
                        "label": "Custom Content WYSIWYG",
                        "name": "custom_content_wysiwyg",
                        "type": "flexible_content",
                        "instructions": "",
                        "required": 1,
                        "conditional_logic": [
                            [
                                {
                                    "field": "field_57e97ff963e85",
                                    "operator": "==",
                                    "value": "WYSIWYG"
                                }
                            ]
                        ],
                        "wrapper": {
                            "width": "",
                            "class": "",
                            "id": ""
                        },
                        "button_label": "Add Row",
                        "min": "",
                        "max": "",
                        "layouts": [
                            {
                                "key": "57e9952363118",
                                "name": "",
                                "label": "",
                                "display": "block",
                                "sub_fields": [
                                    {
                                        "key": "field_57f2d78e751b7",
                                        "label": "WYSIWYG Content",
                                        "name": "",
                                        "type": "tab",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "placement": "top",
                                        "endpoint": 0
                                    },
                                    {
                                        "key": "field_57e995312bd40",
                                        "label": "Title",
                                        "name": "ccw_title",
                                        "type": "text",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "default_value": "",
                                        "placeholder": "",
                                        "prepend": "",
                                        "append": "",
                                        "maxlength": "",
                                        "readonly": 0,
                                        "disabled": 0
                                    },
                                    {
                                        "key": "field_57e995452bd41",
                                        "label": "WYSIWYG",
                                        "name": "ccw_wysiwyg",
                                        "type": "wysiwyg",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "default_value": "",
                                        "tabs": "all",
                                        "toolbar": "full",
                                        "media_upload": 1
                                    },
                                    {
                                        "key": "field_57f2d7a1751b8",
                                        "label": "WYSIWYG Options",
                                        "name": "",
                                        "type": "tab",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "placement": "top",
                                        "endpoint": 0
                                    },
                                    {
                                        "key": "field_57f2d7b2751b9",
                                        "label": "WYSIWYG Alignment",
                                        "name": "ccw_alignment",
                                        "type": "select",
                                        "instructions": "Choose alignment",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "choices": {
                                            "default": "default",
                                            "aligncenter": "aligncenter",
                                            "alignright": "alignright"
                                        },
                                        "default_value": [],
                                        "allow_null": 0,
                                        "multiple": 0,
                                        "ui": 0,
                                        "ajax": 0,
                                        "return_format": "value",
                                        "placeholder": ""
                                    },
                                    {
                                        "key": "field_57f2d7d2751ba",
                                        "label": "WYSIWYG Width",
                                        "name": "ccw_width",
                                        "type": "select",
                                        "instructions": "Want the width to be smaller than 100%?",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "choices": {
                                            "default": "default",
                                            "unit-8-12": "unit-8-12",
                                            "unit-6-12": "unit-6-12",
                                            "unit-4-12": "unit-4-12"
                                        },
                                        "default_value": [
                                            "default"
                                        ],
                                        "allow_null": 0,
                                        "multiple": 0,
                                        "ui": 0,
                                        "ajax": 0,
                                        "return_format": "value",
                                        "placeholder": ""
                                    },
                                    {
                                        "key": "field_57f2d929a6774",
                                        "label": "WYSIWYG Padding",
                                        "name": "ccw_padding",
                                        "type": "select",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "choices": {
                                            "padded": "padded",
                                            "default": "default"
                                        },
                                        "default_value": [],
                                        "allow_null": 0,
                                        "multiple": 0,
                                        "ui": 0,
                                        "ajax": 0,
                                        "return_format": "value",
                                        "placeholder": ""
                                    }
                                ],
                                "min": "",
                                "max": ""
                            }
                        ]
                    },
                    {
                        "key": "field_57e9820de11c0",
                        "label": "Custom Content Icon And Content",
                        "name": "custom_content_icon_and_content",
                        "type": "flexible_content",
                        "instructions": "",
                        "required": 0,
                        "conditional_logic": [
                            [
                                {
                                    "field": "field_57e97ff963e85",
                                    "operator": "==",
                                    "value": "Icon & Content"
                                }
                            ]
                        ],
                        "wrapper": {
                            "width": "",
                            "class": "",
                            "id": ""
                        },
                        "button_label": "Add Icon & Content",
                        "min": "",
                        "max": "",
                        "layouts": [
                            {
                                "key": "57e9838c08001",
                                "name": "",
                                "label": "",
                                "display": "block",
                                "sub_fields": [
                                    {
                                        "key": "field_57e9868fb474e",
                                        "label": "Text",
                                        "name": "",
                                        "type": "tab",
                                        "instructions": "",
                                        "required": "",
                                        "conditional_logic": "",
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "placement": "top",
                                        "endpoint": 0
                                    },
                                    {
                                        "key": "field_57e987c9ca5bf",
                                        "label": "Title",
                                        "name": "cciac_title",
                                        "type": "text",
                                        "instructions": "",
                                        "required": "",
                                        "conditional_logic": "",
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "default_value": "",
                                        "placeholder": "",
                                        "prepend": "",
                                        "append": "",
                                        "maxlength": "",
                                        "readonly": 0,
                                        "disabled": 0
                                    },
                                    {
                                        "key": "field_57e983f5f626a",
                                        "label": "Text",
                                        "name": "cciac_text",
                                        "type": "textarea",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "default_value": "",
                                        "placeholder": "",
                                        "maxlength": "",
                                        "rows": "",
                                        "new_lines": "wpautop",
                                        "readonly": 0,
                                        "disabled": 0
                                    },
                                    {
                                        "key": "field_57e9867eb474d",
                                        "label": "Icon",
                                        "name": "",
                                        "type": "tab",
                                        "instructions": "",
                                        "required": 1,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "placement": "top",
                                        "endpoint": 0
                                    },
                                    {
                                        "key": "field_57e9869cb474f",
                                        "label": "Icon",
                                        "name": "cciac_icon",
                                        "type": "select",
                                        "instructions": "",
                                        "required": 1,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "choices": {
                                            "world": "world",
                                            "share": "share",
                                            "organize": "organize",
                                            "mine": "mine",
                                            "rarecode": "rarecode",
                                            "carriercode": "carriercode",
                                            "random": "random",
                                            "interpretation": "interpretation",
                                            "report": "report",
                                            "pipe": "pipe",
                                            "drag": "drag",
                                            "cloud": "cloud",
                                            "collab": "collab",
                                            "lock": "lock",
                                            "lightning": "lightning",
                                            "scissors": "scissors",
                                            "checkmark": "checkmark",
                                            "brackets": "brackets",
                                            "dna": "dna",
                                            "data": "data",
                                            "hospital": "hospital",
                                            "institution": "institution",
                                            "science": "science",
                                            "team": "team",
                                            "settings": "settings",
                                            "dr--female": "dr--female",
                                            "dr--male": "dr--male",
                                            "person--business--male": "person--business--male",
                                            "person--female": "person--female",
                                            "patients": "patients",
                                            "clinical": "clinical",
                                            "sequencing": "sequencing",
                                            "biotech": "biotech",
                                            "discovery": "discovery"
                                        },
                                        "default_value": [],
                                        "allow_null": 0,
                                        "multiple": 0,
                                        "ui": 0,
                                        "ajax": 0,
                                        "placeholder": "",
                                        "disabled": 0,
                                        "readonly": 0,
                                        "return_format": "value"
                                    },
                                    {
                                        "key": "field_57e983b4f6269",
                                        "label": "Icon Placement",
                                        "name": "cciac_icon_placement",
                                        "type": "radio",
                                        "instructions": "Pick which side the icon goes on",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "choices": {
                                            "Left": "Left",
                                            "Right": "Right"
                                        },
                                        "allow_null": 0,
                                        "other_choice": 0,
                                        "save_other_choice": 0,
                                        "default_value": "",
                                        "layout": "horizontal",
                                        "return_format": "value"
                                    }
                                ],
                                "min": "",
                                "max": ""
                            }
                        ]
                    },
                    {
                        "key": "field_57e98d4fcb96f",
                        "label": "Custom Content Image w\/Caption",
                        "name": "custom_content_image_with_caption",
                        "type": "flexible_content",
                        "instructions": "",
                        "required": 1,
                        "conditional_logic": [
                            [
                                {
                                    "field": "field_57e97ff963e85",
                                    "operator": "==",
                                    "value": "Image w\/ Caption"
                                }
                            ]
                        ],
                        "wrapper": {
                            "width": "",
                            "class": "",
                            "id": ""
                        },
                        "button_label": "Add Row",
                        "min": "",
                        "max": "",
                        "layouts": [
                            {
                                "key": "57e98db15ba03",
                                "name": "",
                                "label": "",
                                "display": "block",
                                "sub_fields": [
                                    {
                                        "key": "field_57e98dbfcb970",
                                        "label": "Image",
                                        "name": "cciwc_image",
                                        "type": "image",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "return_format": "array",
                                        "preview_size": "thumbnail",
                                        "library": "all",
                                        "min_width": "",
                                        "min_height": "",
                                        "min_size": "",
                                        "max_width": "",
                                        "max_height": "",
                                        "max_size": "",
                                        "mime_types": ".jpg"
                                    },
                                    {
                                        "key": "field_57e98df8cb971",
                                        "label": "Caption",
                                        "name": "cciwc_caption",
                                        "type": "text",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "default_value": "",
                                        "placeholder": "",
                                        "prepend": "",
                                        "append": "",
                                        "maxlength": "",
                                        "readonly": 0,
                                        "disabled": 0
                                    }
                                ],
                                "min": "",
                                "max": ""
                            }
                        ]
                    },
                    {
                        "key": "field_57e98e5dcb972",
                        "label": "Custom Content Button",
                        "name": "custom_content_button",
                        "type": "flexible_content",
                        "instructions": "",
                        "required": 1,
                        "conditional_logic": [
                            [
                                {
                                    "field": "field_57e97ff963e85",
                                    "operator": "==",
                                    "value": "Button"
                                }
                            ]
                        ],
                        "wrapper": {
                            "width": "",
                            "class": "",
                            "id": ""
                        },
                        "button_label": "Add Row",
                        "min": "",
                        "max": "",
                        "layouts": [
                            {
                                "key": "57e98e6a783a7",
                                "name": "",
                                "label": "",
                                "display": "block",
                                "sub_fields": [
                                    {
                                        "key": "field_57e98e6ccb973",
                                        "label": "Button Text",
                                        "name": "ccb_button_text",
                                        "type": "text",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "default_value": "",
                                        "placeholder": "",
                                        "prepend": "",
                                        "append": "",
                                        "maxlength": "",
                                        "readonly": 0,
                                        "disabled": 0
                                    },
                                    {
                                        "key": "field_57e98e88cb974",
                                        "label": "Button URL",
                                        "name": "ccb_button_url",
                                        "type": "text",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "default_value": "",
                                        "placeholder": "",
                                        "prepend": "",
                                        "append": "",
                                        "maxlength": "",
                                        "readonly": 0,
                                        "disabled": 0
                                    },
                                    {
                                        "key": "field_57e98e9acb975",
                                        "label": "Button Style",
                                        "name": "ccb_button_style",
                                        "type": "select",
                                        "instructions": "Desire something different than the normal blue button? Pick another",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "choices": {
                                            "default": "default",
                                            "white": "white",
                                            "red": "red"
                                        },
                                        "default_value": [],
                                        "allow_null": 0,
                                        "multiple": 0,
                                        "ui": 0,
                                        "ajax": 0,
                                        "placeholder": "",
                                        "disabled": 0,
                                        "readonly": 0,
                                        "return_format": "value"
                                    }
                                ],
                                "min": "",
                                "max": ""
                            }
                        ]
                    },
                    {
                        "key": "field_57e990add1d40",
                        "label": "Custom Content 50\/50 List",
                        "name": "custom_content_5050_list",
                        "type": "flexible_content",
                        "instructions": "",
                        "required": 1,
                        "conditional_logic": [
                            [
                                {
                                    "field": "field_57e97ff963e85",
                                    "operator": "==",
                                    "value": "50\/50 List"
                                }
                            ]
                        ],
                        "wrapper": {
                            "width": "",
                            "class": "",
                            "id": ""
                        },
                        "button_label": "Add Row",
                        "min": "",
                        "max": "",
                        "layouts": [
                            {
                                "key": "57e990c9759c5",
                                "name": "",
                                "label": "",
                                "display": "block",
                                "sub_fields": [
                                    {
                                        "key": "field_57e9927783b25",
                                        "label": "Left List",
                                        "name": "left_list",
                                        "type": "tab",
                                        "instructions": "",
                                        "required": "",
                                        "conditional_logic": "",
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "placement": "top",
                                        "endpoint": 0
                                    },
                                    {
                                        "key": "field_57e990cbd1d41",
                                        "label": "Left Title",
                                        "name": "ccffl_left_title",
                                        "type": "text",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "default_value": "",
                                        "placeholder": "",
                                        "prepend": "",
                                        "append": "",
                                        "maxlength": "",
                                        "readonly": 0,
                                        "disabled": 0
                                    },
                                    {
                                        "key": "field_57e990ddd1d42",
                                        "label": "Left List Items",
                                        "name": "ccffl_left_list_items",
                                        "type": "repeater",
                                        "instructions": "",
                                        "required": 0,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "collapsed": "",
                                        "min": "",
                                        "max": "",
                                        "layout": "table",
                                        "button_label": "Add Row",
                                        "sub_fields": [
                                            {
                                                "key": "field_57e990edd1d43",
                                                "label": "List Item",
                                                "name": "ccffl_list_item",
                                                "type": "text",
                                                "instructions": "",
                                                "required": 0,
                                                "conditional_logic": 0,
                                                "wrapper": {
                                                    "width": "",
                                                    "class": "",
                                                    "id": ""
                                                },
                                                "default_value": "",
                                                "placeholder": "",
                                                "prepend": "",
                                                "append": "",
                                                "maxlength": "",
                                                "readonly": 0,
                                                "disabled": 0
                                            }
                                        ]
                                    },
                                    {
                                        "key": "field_57e9928983b26",
                                        "label": "Right List",
                                        "name": "right_list",
                                        "type": "tab",
                                        "instructions": "",
                                        "required": "",
                                        "conditional_logic": "",
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "placement": "top",
                                        "endpoint": 0
                                    },
                                    {
                                        "key": "field_57e99125d1d44",
                                        "label": "Right List Title",
                                        "name": "ccffl_right_list_title",
                                        "type": "text",
                                        "instructions": "",
                                        "required": "",
                                        "conditional_logic": "",
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "default_value": "",
                                        "placeholder": "",
                                        "prepend": "",
                                        "append": "",
                                        "maxlength": "",
                                        "readonly": 0,
                                        "disabled": 0
                                    },
                                    {
                                        "key": "field_57e9912ed1d45",
                                        "label": "Right List Items",
                                        "name": "ccffl_right_list_items",
                                        "type": "repeater",
                                        "instructions": "",
                                        "required": 1,
                                        "conditional_logic": 0,
                                        "wrapper": {
                                            "width": "",
                                            "class": "",
                                            "id": ""
                                        },
                                        "collapsed": "",
                                        "min": "",
                                        "max": "",
                                        "layout": "table",
                                        "button_label": "Add Row",
                                        "sub_fields": [
                                            {
                                                "key": "field_57e99154d1d46",
                                                "label": "Right List Item",
                                                "name": "ccffl_right_list_item",
                                                "type": "text",
                                                "instructions": "",
                                                "required": 0,
                                                "conditional_logic": 0,
                                                "wrapper": {
                                                    "width": "",
                                                    "class": "",
                                                    "id": ""
                                                },
                                                "default_value": "",
                                                "placeholder": "",
                                                "prepend": "",
                                                "append": "",
                                                "maxlength": "",
                                                "readonly": 0,
                                                "disabled": 0
                                            }
                                        ]
                                    }
                                ],
                                "min": "",
                                "max": ""
                            }
                        ]
                    }
                ]
            }
        ],
        "location": [
            [
                {
                    "param": "post_type",
                    "operator": "==",
                    "value": "capability"
                }
            ]
        ],
        "menu_order": 0,
        "position": "normal",
        "style": "default",
        "label_placement": "top",
        "instruction_placement": "label",
        "hide_on_screen": "",
        "active": 1,
        "description": ""
    }
]`



PS looks like @germanny had a similar issue again: https://support.advancedcustomfields.com/forums/topic/data-from-flexible-content-fields-inside-repeater-fields-is-left-hanging/ but ...since April?

**Comment by elliotcondon** [comment]
Hi @ukeboyska 

Thanks for the attachment. I've imported your field group and done some tests, but I am not able to replicate your issue. If I delete a layout, or a repeater row, all the data is removed from the DB.

I have checked the database before and after each action to confirm, and everything works as expected.

1. Can you please ensure you have updated to the latest version of ACF PRO and test again?
2. Please consider re-thinking your field group design. Your current setup is very complicated to achieve such as basic page builder. I suggest removing the parent repeater field completely, and using the flexible content field as it should be used (with multiple layouts to chose from)

**Comment by ukeboyska** [comment]
thanks again! i had a feeling i was over-complicating things...hadn't discovered the multiple layout feature yet in flexible content (duh/it's awesome!) hopefully that alone will fix my issue.
